zookeeper/source/zk_nn_client.cc

```diff
@@ -1552,6 +1733,50 @@ bool ZkNnClient::rename_ops_for_file(const std::string &src,
     LOG(INFO) << "Added op#" << ops.size() << ": create " << dst_znode_blocks;
     ops.push_back(zk->build_create_op(dst_znode_blocks, data));
 
+  // Copy over the lease branch
+  auto lease_data = std::vector<std::uint8_t>();
+  zk->get(src_znode_leases, lease_data, error_code);
+  if (error_code != ZOK) {
+    LOG(ERROR) << "Failed to get data from '"
+    << src_znode_leases
+    << "' when renaming.";
+    return false;
+  }
+  ops.push_back(zk->build_create_op(dst_znode_leases, lease_data));
```

Comment:
Not sure if it's worth doing the same thing here, but we should probably modularize out this method of constructing leases.